sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -185,16 +187,44 @@ public static BinaryData fromBytes(byte[] data) {
         return new BinaryData(data);
     }
 
+    /**
+     * Serialize the given {@link Object} into {@link BinaryData} using json serializer which is available in classpath.
+     * The serializer must implement {@link JsonSerializer} interface. A singleton instance of {@link JsonSerializer}
+     * is kept for this class to use. If {@code null} data is provided , it will be converted into empty byte array.
+     *
+     * @param data The {@link Object} which needs to be serialized into bytes.
+     * @throws IllegalStateException If a {@link JsonSerializer} cannot be found on the classpath.
+     * @return {@link BinaryData} representing binary data.
+     *
+     * @see JsonSerializer
+     */
+    public static BinaryData fromObject(Object data) {
```

Comment:
Given this API is converting into JSON we may want to call it `fromObjectJson`.

Reply:
If we change it, we need to sync with .Net. I copied this API name from them.